### PR TITLE
#755 fixed; Updated integration tests.

### PR DESCRIPTION
**Pull request by ognjenns**

### CODE REVIEW

File: Examples/restcomm-olympus/app/src/androidTest/java/org/restcomm/android/olympus/IntegrationTests.java [hunk 1]
```diff
@@ -1531,7 +1458,7 @@ public void run() {
             }
         });
 
-        await().atMost(PUSH_TIMEOUT, TimeUnit.SECONDS).until(fieldIn(this).ofType(boolean.class).andWithName("onWarningReceived"), equalTo(true));
+        await().atMost(PUSH_TIMEOUT, TimeUnit.SECONDS).until(fieldIn(this).ofType(boolean.class).andWithName("deviceInitialized"), equalTo(true));
```
Comment (atsakiridis) [inline Examples/restcomm-olympus/app/src/androidTest/java/org/restcomm/android/olympus/IntegrationTests.java]: Can introduce a new state for deviceReconfigured and change deviceInitialized -> deviceReconfigured

File: restcomm.android.sdk/src/main/java/org/restcomm/android/sdk/RCDevice.java [hunk 2]
```diff
@@ -313,7 +313,7 @@
    public static String EXTRA_VIDEO_ENABLED = "org.restcomm.android.sdk.VIDEO_ENABLED";
 
    // Notification ids for calls and messages. Key is the sender and value in the notification id. We mainly need those
-   // to make sure that notifications from a specific user are shown in a single slot, to avoid confusing the user. This means
+   // to make sure that notifications from a specific user are shown onReconfiguredin a single slot, to avoid confusing the user. This means
```
Comment (atsakiridis) [inline restcomm.android.sdk/src/main/java/org/restcomm/android/sdk/RCDevice.java]: Can you elaborate on onReconfiguredin? Is it a typo?

File: restcomm.android.sdk/src/main/java/org/restcomm/android/sdk/RCDevice.java [hunk 3]
```diff
@@ -1128,8 +1128,8 @@ public void onCloseReply(String jobId, RCClient.ErrorCodes status, String text)
    {
       RCLogger.i(TAG, "onCloseReply(): id: " + jobId + ", status: " + status + ", text: " + text);
 
-      if (listener != null) {
-         listener.onReleased(this, status.ordinal(), text);
+      if (this.listener != null) {
```
Comment (atsakiridis) [inline restcomm.android.sdk/src/main/java/org/restcomm/android/sdk/RCDevice.java]: Lost you, can you elaborate why we need to add this. in all these occurrences?

File: restcomm.android.sdk/src/main/java/org/restcomm/android/sdk/RCDevice.java (hunk 2 above)
Comment (ognjenns) [reply restcomm.android.sdk/src/main/java/org/restcomm/android/sdk/RCDevice.java]: typo

File: restcomm.android.sdk/src/main/java/org/restcomm/android/sdk/RCDevice.java (hunk 3 above)
Comment (ognjenns) [reply restcomm.android.sdk/src/main/java/org/restcomm/android/sdk/RCDevice.java]: in RCDevice, we have in some places listener and in some places this.listener. I just want it to make even.

**Pull request merged**